scripts/finish-release.ts

```diff
@@ -0,0 +1,244 @@
+import { execSync } from 'child_process';
+import * as rl from 'readline';
+import { stdin, stdout } from 'process';
+import yargs from 'yargs';
+
+export function extractUpstreamNameFromRemotes(remoteGitVerbose: string, upstreamRepositoryName: string): string | undefined {
+  const lineWithRepoName = remoteGitVerbose.split('\n').find((l) => l.indexOf(upstreamRepositoryName) > -1);
+  return lineWithRepoName?.split(/\s/)?.[0];
+}
+
+export interface MergeOptions {
+  message: string;
+  mode: 'ff-only';
+}
+export class Git {
+  remote(verbose?: boolean): string {
+    let command = ['git', 'remote'];
+    if (verbose) {
+      command.push('-v');
+    }
+    return execSync(command.join(' ')).toString();
```

Comment:
for all of these I would prefer to use execaSync from https://www.npmjs.com/package/execa (we already have dependency on this package). That way we don't have to worry about args with spaces in them in the future (which would break this logic)